python/Ice/greeter/server_asyncio/main.py

```diff
@@ -0,0 +1,46 @@
+#!/usr/bin/env python
+# Copyright (c) ZeroC, Inc.
+
+import Ice
+import chatbot
+import asyncio
+
+
+async def main():
+
+    loop = asyncio.get_event_loop()
+
+    def coroutineExecutor(coroutine):
+        """
+        A custom coroutine executor that runs the coroutine in the main asyncio event loop.
+        """
+
+        # We use run_coroutine_threadsafe because the executor is called by an Ice thread.
+        return asyncio.run_coroutine_threadsafe(coroutine, loop)
+
+    # Create an initialization data structure that specifies the coroutine executor.
+    initData = Ice.InitializationData()
+    initData.coroutineExecutor = coroutineExecutor
+
+    # Create an Ice communicator to initialize the Ice runtime. The communicator is disposed before the program exits.
```

Comment:
"disposed" is a C# term, and the scope is not the program but the `with block`. What about:

> The communicator is destroyed at the end of the with block.